src/classes/UTIL_ACCT_Naming.cls

```diff
@@ -35,296 +35,647 @@
+    }
+
+    /*******************************************************************************************************
+    * @description This sets the default Account name for each respective Account model.
+    * @param cons a list of Contacts whose Account needs Name update.
+    * @param accountNamingFormat is a string of the Account name format currently set in the org.
+    * @return String.
+    */
+    private static String defaultAccountName(List<Contact> cons, String accountNamingFormat) {
+        Set<String> householdLastNames = new Set<String>();
+        String finalAccountName;
+
+        for (Contact con : cons) {
+            if (defaultRecTypeId == hhAccountRecordTypeId) {
                 householdLastNames.add(con.LastName);
+            } else if (defaultRecTypeId == adminAccountRecordTypeId) {
+                finalAccountName = con.LastName + ' ' + System.label.DefaultAdminName;
+                break;
+            } else {
+                finalAccountName = con.LastName + ' ' + System.label.DefaultAccountName;
+                break;
             }
+        }
+
+        if (householdLastNames.size() > 0) {
             String name = '';
+            String strAnd = ' ' + Label.defaultNamingConnector + ' ';
             Integer counter = 0;
+
             for (String lastName : householdLastNames) {
                 name += lastName;
                 counter++;
                 if (counter < householdLastNames.size()) {
-                    name += ' ' + Label.defaultNamingConnector +' ';
+                    name += strAnd;
                 }
             }
-            return name + ' ' + System.label.DefaultHouseholdName;
+
+            if (accountNamingFormat == hhNameFormat) {
+                finalAccountName =  name + ' ' + System.label.DefaultHouseholdName;
+            }
         }
+
+        return finalAccountName;
     }
 
+    private static String chLToken = '{!';
+    private static String chRToken = '}';
+    public static Boolean setFNSpec = false;
+
     /*******************************************************************************************************
-    * @description returns the default name for a new Administrative Account.
-    * @param con the Contact
-    * @return string. the Admin account name
+    * @description Class that supports the parsing of Account name format set in the current org and Account model.
+    * NameSpec looks like: Prefix {LastName} {{FirstName}} Suffix
+    * firstNameSpec is a combo of fields and literals like: {Salutation} {FirstName} Family
+    * @return NameSpec.
     */
-    public static String strNameAdmAccountForContact(Contact con) {
-        String adminAccNamingFormat = UTIL_CustomSettingsFacade.getSettings().Admin_Account_Naming_Format__c;
-        if (adminAccNamingFormat == Label.acctNamingOther) {
-            adminAccNamingFormat = UTIL_CustomSettingsFacade.getSettings().Admin_Other_Name_Setting__c;
+    private class NameSpec {
```

Comment:
I didn't really review must of this class and its inner methods... at least, not functionality-wise. Is this what was mostly copied from NPSP?

Reply:
yep, this is it. This essentially pulls and breaks out the selected Account name format that will be used later on in the buildAccountName() method.